OpenOCD configuration fixes for ARC

IRISZZW opened the pull request:
In newer OpenOCD version from Zephyr's SDK v0.12, there are some
changes in OpenOCD scripts.

- OpenOCD cofig command: ftdi_device_desc description provides the
USB device description (the iProduct string) of the adapter. And If not
specified, the device description is ignored during device selection.
In file interface/ftdi/digilent-hs1.cfg, ftdi_device_desc will be set to
"Digilent Adept USB Device", while we get the iProduct string
"Digilent USB Device" from em_starterkit adapter.  it's better not
specify it and only use the vid and pid of the adapter for selection.

- there are some changes in OpenOCD scripts: JTAG probe interface 
(AKA "adapter") setup, see http://openocd.zylin.com/#/c/5784/

So we need to change OpenOCD scripts accordingly to match
newer OpenOCD version from Zephyr's SDK v0.12. 

Note: the fix for HSDK board is done in PR: #30266

IRISZZW committed:
board: arc: em_starterkit*: fix OpenOCD configuration ftdi_device_desc

OpenOCD cofig command: ftdi_device_desc description provides the
USB device description (the iProduct string) of the adapter. And If not
specified, the device description is ignored during device selection.

In newer OpenOCD version from Zephyr's SDK v0.12, there are some
changes in OpenOCD scripts.
In file interface/ftdi/digilent-hs1.cfg, ftdi_device_desc will be set to
"Digilent Adept USB Device", while we get the iProduct string
"Digilent USB Device" from em_starterkit adapter.  it's better not
specify it and only use the vid and pid of the adapter for selection.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

abrodkin commented on boards/arc/em_starterkit/support/openocd.cfg:
```diff
@@ -1,6 +1,12 @@
 # Configure JTAG cable
 # EM Starter Kit has built-in FT2232 chip, which is similar to Digilent HS-1.
-source [find interface/ftdi/digilent-hs1.cfg]
+interface ftdi
```
If you're updating this to accommodate newer OpenOCD please make sure you follow suggested renames (`interface` -> `adapter driver`) etc. Look at OpenOCD's debug messages where all that is explained.

Note, other options might need update as well.

abrodkin reviewed (changes requested):
The change itself is good, but please update OpenOCD options according to its recommendations (might be seen from OpenOCD's debug messages).

IRISZZW committed:
board: arc: accommodate upstream OpenOCD for ARC

In newer OpenOCD version from Zephyr's SDK v0.12, there are some
changes in OpenOCD scripts: JTAG probe interface (AKA "adapter")
setup, see http://openocd.zylin.com/#/c/5784/

And so we need to change OpenOCD scripts accordingly to match
newer OpenOCD version from Zephyr's SDK v0.12.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

abrodkin reviewed (approved):
LGTM now, thanks!
